### PR TITLE
[PGPNP] Add support for MySql and MSSql to Signer

### PR DESCRIPTION
**Pull request by jmrossy**
### Description

Add support for postgres db alternatives
Initialize and test DB connection on service start

### Tested

Ran services against MySql in GCP staging and MS SQL in Azure staging

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/phone-number-privacy/signer/src/database/database.ts [hunk 1]
```diff
@@ -1,16 +1,88 @@
...
-})
+let db: Knex
+export async function initDatabase(doTestQuery = true) {
+  logger.info('Initializing database connection')
+  const { type, host, port, user, password, database, ssl } = config.db
 
-export function getDatabase() {
+  let dbConfig: any
+  let client: string
+  if (type === SupportedDatabase.Postgres) {
```
Comment (codyborn) [inline packages/phone-number-privacy/signer/src/database/database.ts:14]: Should this be a case insensitive comparison? No strong preference but wanted to be explicit.

File: packages/phone-number-privacy/signer/src/database/database.ts [hunk 2]
```diff
@@ -1,16 +1,88 @@
...
+  }
+
+  db = knex({
+    client,
+    connection: dbConfig,
+    debug: DEV_MODE,
+  })
+
+  if (doTestQuery) {
+    await executeTestQuery(db)
```
Comment (codyborn) [inline packages/phone-number-privacy/signer/src/database/database.ts:57]: nice!

**Review by codyborn — APPROVED** [review]
Looks good! Would be great to add some simple unit tests before checking in.

File: packages/phone-number-privacy/signer/src/database/database.ts (hunk 1 above)
Comment (jmrossy) [reply packages/phone-number-privacy/signer/src/database/database.ts:14]: We lower case the value as it gets put in the config so that's covered

**Pull request merged**